### DISCUSSION TITLE
Is it possible for the LGT8F328P to act as an RS422 / RS485 transceiver?

### DISCUSSION THREAD

**Discussion by LaZsolt** [dbuezas/lgt8fx#326] [Ideas]
Is it possible to use a special version of SoftwareSerial to turn the LGT8F328P into an RS422 / RS485 transreceiver?
The two output bits (TX+ and TX-) can be written on any port at same time by software.
According to my imagination the input lines (RX+ and RX-) received through analog comparator.

**Comment by dbuezas** [comment]
And what do you do about the floating gnd?
The onboard analog comparator only works when both lines are within the rails

**Reply by LaZsolt** [reply]
RS422 wiring:
![kép](https://github.com/user-attachments/assets/cc0a205c-dec8-4da8-9f6e-d95578e8239a)

There is no floating gnd.

**Reply by dbuezas** [reply]
Oh, good, assumed it didn't have a gnd wire. But aren't the complementary lines symmetric wrt gnd? I.e one line will have negative voltage

**Reply by LaZsolt** [reply]
No, the complementary lines not symmetric to gnd. They always positive to the gnd.

**Reply by dbuezas** [reply]
Then I think you are up to something.
